File: CodeR.Java.Parser/src/java/CodeR.Java/Tool.java
package CodeR.Java;

public class Tool {
    private JavaFileProcessor processor;

    public static void main(String args[]) {
        Tool tool = new Tool(new DefaultJavaFileProcessor());

        tool.run(args);

    }

    public Tool(JavaFileProcessor processor) {

        this.processor = processor;
    }

    public void run(String[] args) {
        for (String arg : args) {
            processor.processFile(arg);
        }
    }

}

File: src/scala/metamorph/java/Tool.scala
package metamorph.java

object Tool {
  def main(args: Array[String]) {
    val tool = new Tool(new DefaultJavaFileProcessor())

    tool.run(args)
  }
}

class Tool(processor: JavaFileProcessor) {
  def run(args: Array[String]) {

    args foreach (arg => processor.processFile(arg))
  }
}
